### PR TITLE
🏷️ [RUMF-1098] move init options into their related interfaces

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

Improve init configuration typings 

## Changes

* 🚚 Move some `InitConfiguration` properties to their own package.
* 🎨 Group/reorder properties

## Testing

- [ ] Local
- [x] Staging
- [ ] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/core/src/domain/configuration/configuration.ts [hunk 1]
```diff
@@ -14,30 +14,29 @@ export const DefaultPrivacyLevel = {
 export type DefaultPrivacyLevel = typeof DefaultPrivacyLevel[keyof typeof DefaultPrivacyLevel]
 
 export interface InitConfiguration {
+  // global options
   clientToken: string
-  applicationId?: string | undefined
-  actionNameAttribute?: string | undefined
-  internalMonitoringApiKey?: string | undefined
-  allowedTracingOrigins?: Array<string | RegExp> | undefined
+  beforeSend?: BeforeSendCallback | undefined
```
Comment (bcaudan) [inline packages/core/src/domain/configuration/configuration.ts:19]: do we need to keep these common `beforeSend` and `BeforeSendCallback`?
cf #822

File: packages/rum-core/src/domain/configuration.ts [hunk 2]
```diff
@@ -12,9 +12,23 @@ import { RumEventDomainContext } from '../domainContext.types'
 import { RumEvent } from '../rumEvent.types'
 
 export interface RumInitConfiguration extends InitConfiguration {
+  // global options
   applicationId: string
   beforeSend?: ((event: RumEvent, context: RumEventDomainContext) => void | boolean) | undefined
+
+  // tracing options
+  allowedTracingOrigins?: Array<string | RegExp> | undefined
```
Comment (bcaudan) [inline packages/rum-core/src/domain/configuration.ts:20]: Should we use `ReadonlyArray` now?

File: packages/core/src/domain/configuration/configuration.ts (hunk 1 above)
Comment (BenoitZugmeyer) [reply packages/core/src/domain/configuration/configuration.ts:19]: Mmh I can move it. The downside is that we'll need to have the following code in both RUM and Logs configuration build:
```ts
    beforeSend:
      initConfiguration.beforeSend && catchUserErrors(initConfiguration.beforeSend, 'beforeSend threw an error:'),
```
but I guess that's fine

Comment (bcaudan) [reply packages/core/src/domain/configuration/configuration.ts:19]: indeed.
Alternate proposal: keep the code here but make more clear here that it is a more generic type with real implementation in logs / rum.
it should avoid the confusion of the related ticket.
wdyt?

File: packages/core/src/domain/configuration/configuration.ts [hunk 3]
```diff
@@ -41,7 +41,9 @@ export interface InitConfiguration {
   replica?: ReplicaUserConfiguration | undefined
 }
 
-export type BeforeSendCallback = (event: any, context?: any) => unknown
+// This type is only used to build the core configuration. Logs and RUM SDKs are using a proper type
+// for this option.
+type GenericBeforeSendCallback = (event: any, context?: any) => unknown
```
Comment (bcaudan) [inline packages/core/src/domain/configuration/configuration.ts:46]: 👍

**Review by bcaudan — APPROVED** [review]

**Review by amortemousque — APPROVED** [review]

**Pull request merged**